Core/include/Acts/Propagator/TryAllNavigator.hpp

```diff
@@ -0,0 +1,477 @@
+                      "Good luck.");
+        continue;
+      }
+
+      ACTS_VERBOSE(volInfo(state) << "aiming at surface "
+                                  << surface.geometryId() << ". step size is "
+                                  << stepper.outputStepSize(state.stepping));
+      break;
+    }
+
+    if (intersectionCandidates.empty()) {
+      stepper.releaseStepSize(state.stepping, ConstrainedStep::actor);
+
+      ACTS_VERBOSE(volInfo(state) << "no intersections found. advance without "
+                                     "constraints. step size is "
+                                  << stepper.outputStepSize(state.stepping));
+    }
+
+    state.navigation.intersectionCandidates = std::move(intersectionCandidates);
+  }
+
+  /// @brief Navigator post step call
+  ///
+  /// This determines if we hit the next navigation candidate and deals with it
+  /// accordingly. It sets the current surface, enters layers and changes
+  /// volumes.
+  ///
+  /// @tparam propagator_state_t is the type of Propagatgor state
+  /// @tparam stepper_t is the used type of the Stepper by the Propagator
+  ///
+  /// @param [in,out] state is the mutable propagator state object
+  /// @param [in] stepper Stepper in use
+  ///
+  /// @return Boolean to indicate if we continue with the actors and
+  ///         aborters or if we should target again.
+  template <typename propagator_state_t, typename stepper_t>
+  void postStep(propagator_state_t& state, const stepper_t& stepper) const {
+    ACTS_VERBOSE(volInfo(state) << "post step");
+
+    assert(state.navigation.currentSurface == nullptr &&
+           "Current surface must be reset.");
+
+    if (state.navigation.intersectionCandidates.empty()) {
+      ACTS_VERBOSE(volInfo(state) << "no intersections.");
+      return;
+    }
+
+    std::vector<detail::IntersectionCandidate> hitCandidates;
+
+    for (const auto& candidate : state.navigation.intersectionCandidates) {
+      const auto& intersection = candidate.intersection;
+      const Surface& surface = *intersection.object();
+
+      Intersection3D::Status surfaceStatus = stepper.updateSurfaceStatus(
+          state.stepping, surface, intersection.index(),
+          state.options.direction, BoundaryCheck(false),
+          state.options.surfaceTolerance, logger());
+
+      if (surfaceStatus != IntersectionStatus::onSurface) {
+        break;
+      }
+
+      hitCandidates.emplace_back(candidate);
+    }
+
+    if (hitCandidates.empty()) {
+      ACTS_VERBOSE(volInfo(state) << "Staying focussed on surface.");
+      return;
+    }
+
+    state.navigation.intersectionCandidates.clear();
+
+    ACTS_VERBOSE(volInfo(state)
+                 << "Found " << hitCandidates.size()
+                 << " intersections on surface without bounds check.");
+
+    std::vector<detail::IntersectionCandidate> trueHitCandidates;
+
+    for (const auto& candidate : hitCandidates) {
```

Comment:
The idea of the switching boundary check is that we might not project onto the surface until we are really on it because of the bending. So this is sort of an edge case handling and to be very general in the surface approach.

My idea with this is that we can run the track finding with this navigator and see how much efficiency we loose because of `Navigator` limitations